web-app/src/components/utils/IsOnlineWrapper.js

```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react'
+import api from '../../api';
+
+class IsOnlineWrapper extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isOnline: true,
+        }
+    }
+
+    updateStatus() {
+        const apiStatus = api.status();
+        this.setState({ isOnline: apiStatus !== 'disconnected' })
+    }
+
+    componentDidMount() {
+        this.updateStatus();
+        this.interval = setInterval(() => this.updateStatus(), 1000);
+    }
+
+    render() {
+        return (
+            <>
+                {this.props.children(this.state.isOnline)}
```

Comment:
Why did you choose this option and not React.cloneElement ?

Reply:
I found this example using context 
https://gist.github.com/jvandenaardweg/58ed91e3c33de0b75a15d38853b23d7d

Reply:
With React.cloneElement, I found that the syntax for using it wasn't explicit.
I got something like
```
<Wrapper>
    <Wrapped prop1={val1} ... propn={valn}> 
</Wrapper> 
```
And then taking this.props.isOnline at Wrapped.
Wasn't clear where it came from..

Context api looks cool. Anyway when needing to use more than one context, the syntax is the same as render props: https://reactjs.org/docs/context.html#consuming-multiple-contexts

I'm okay if we go with context api.

Reply:
I think a single context containing all the configuration and status of the app would be enough.

Reply:
Ok, later on we can make the context general, please check if the new implementation looks good.